core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java

```diff
@@ -65,11 +67,10 @@
 
   private ScannerImpl.Reporter reporter;
 
-  private static ThreadPoolExecutor readaheadPool =
-      ThreadPools.createThreadPool(0, Integer.MAX_VALUE, 3L, SECONDS,
-          "Accumulo scanner read ahead thread", new SynchronousQueue<>(), true);
+  private final ThreadPoolExecutor readaheadPool;
```

Comment:
Same as my previous comment, this could be moved to ClientContext and cleaned up on `client.close()`. No need for the Cleaner.

Reply:
Addressed in ec0e7cf